pkgs/by-name/en/enzyme/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  llvmPackages,
+  git,
+}:
+stdenv.mkDerivation rec {
+  pname = "enzyme";
+  version = "0.0.165";
+
+  src = fetchFromGitHub {
+    owner = "EnzymeAD";
+    repo = "Enzyme";
+    rev = "v${version}";
+    hash = "sha256-EyIpbcCJHj09Aa/NMr9BqOHvaWvaRqetCQRy2oxiJKE=";
+  };
+
+  postPatch = ''
+    patchShebangs enzyme
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    git
+  ];
+
+  buildInputs = with llvmPackages; [
+    llvm
+    libclang
+  ];
+
+  cmakeFlags = [
+    "-S../enzyme"
+    "-DLLVM_DIR=${llvmPackages.llvm.dev}"
+    "-DClang_DIR=${llvmPackages.libclang.dev}"
```

Comment:
Not sure if this would work better if you replace `stdenv` with `clangStdenv` and remove all other llvm stuff

Reply:
I'll give this a shot. I thought I tried this already, but perhaps not.

Reply:
Ah right, so the deal is I don't need to build with clang - I just need to reuse the llvm compiler infrastructure. For example, I don't think it matters if we build with gcc, we just need to point to an instance of llvm and libclang. afaik clangStdenv doesn't provide those? https://github.com/NixOS/nixpkgs/issues/277564

Reply:
But tbf I am quite confused on all of the different ways clang/llvm are packaged

Reply:
I have updated to build with clang (which then provides libclang, but not lllvm). I also enabled parallel building.